scripts-dev/lint.sh

```diff
@@ -1,19 +1,11 @@
 #! /usr/bin/env bash
 set -ex
 
-# Keep this up to date with the CI config at .github/workflows/pipeline.yml
+# We don't list explicit directories here. Instead, rely on the tools' default behaviour
+# (or explicit configuration in setup.cfg/pyproject.toml). This helps to keep this script
+# consistent with CI.
 
-black sydent/ stubs/ tests/ matrix_is_test/ scripts/ setup.py
-flake8 sydent/ tests/ matrix_is_test/ scripts/ setup.py
-
-# There's a different convention for formatting stub files.
-# Ignore various error codes from pycodestyle that we don't want
-# to enforce for stubs. (We rely on `black` to format stubs.)
-
-# E301, E302 and E305 complain about missing blank lines.
-# E701 and E7044 complains when we define a function or class entirely within
-# one line.
-flake8 stubs/ --ignore E301,E302,E305,E701,E704
-
-isort sydent/ stubs/ tests/ matrix_is_test/ scripts/ setup.py
+black .
+flake8
+isort .
```

Comment:
How smart are the `.` parameters to these — will they descend into ignored dirs etc?

I suppose there's no longer such a question as 'will it descend into venvs' because Poetry venvs are somewhere else?

Happy to roll with this, after all.. don't mind me.

Reply:
I think you can ask poetry to use an existing virtualenv or to create one in a specific location.

> How smart are the . parameters to these — will they descend into ignored dirs etc?

`black --verbose` skips .gitignored directories. To be honest, I haven't tried the rest recently but I _think_ they did sensible things when I was prototyping this a few weeks back.

> 'will it descend into venvs' 

I did have this problem because one prototype of the setup-poetry action was putting the venv in `.venv` in the checkout. Let me double check that's not still the case...

Reply:
Yeah, the poetry venv is in `path: /home/runner/.cache/pypoetry/virtualenvs` (from the CI logs)